src/java.security.jgss/share/native/libj2gss/NativeUtil.c

```diff
@@ -580,76 +590,128 @@ void checkStatus(JNIEnv *env, jobject jstub, OM_uint32 major,
  * Utility routine for initializing gss_buffer_t structure
  * with the byte[] in the specified jbyteArray object.
  * NOTE: must call resetGSSBuffer() to free up the resources
- * inside the gss_buffer_t structure.
  */
 void initGSSBuffer(JNIEnv *env, jbyteArray jbytes,
-                     gss_buffer_t cbytes) {
-
-  int len;
+                     gss_buffer_t cbytes, jboolean wantCopy)
+{
+  jboolean isCopy;
+  jint len;
   void* value;
 
-  if (jbytes != NULL) {
-    len = (*env)->GetArrayLength(env, jbytes);
-    value = malloc(len);
-    if (value == NULL) {
+  cbytes->length = 0;
+  cbytes->value = NULL;
+
+  if (jbytes == NULL ||
+      (len = (*env)->GetArrayLength(env, jbytes)) == 0)
+    return;
+
+  cbytes->length = len;
+
+  if (wantCopy == JNI_FALSE) {
+    cbytes->value = (*env)->GetByteArrayElements(env, jbytes, &isCopy);
```

Comment:
If you don't plan to read `isCopy`, just pass in `NULL`.

Reply:
... and what will you do if `GetByteArrayElements` doesn't return a copy to you??

Reply:
I think this whole function is unnecessarily complicated. In practice, any modern JVM will return a copy from GetByteArrayElements, anyway. So what do you gain from having two codepaths here?